content_scripts/ui_component.coffee

```diff
@@ -39,13 +60,21 @@ class UIComponent
     @iframeElement.focus()
 
   show: (message) ->
+    unless @loaded
+      @queueAction "show", arguments
+      return
+
     @postMessage message if message?
     @iframeElement.classList.remove "vimiumUIComponentHidden"
-    @iframeElement.classList.add "vimiumUIComponentShowing"
+    @iframeElement.classList.add "vimiumUIComponentVisible"
     @showing = true
 
-  hide: (focusWindow = true)->
-    @iframeElement.classList.remove "vimiumUIComponentShowing"
+  hide: (focusWindow = true, forceImmediate = false) ->
+    unless @loaded or forceImmediate
+      @queueAction "hide", arguments
+      return
+
+    @iframeElement.classList.remove "vimiumUIComponentVisible"
     @iframeElement.classList.add "vimiumUIComponentHidden"
     window.focus() if focusWindow
```

Comment:
> Is this fixable?

@smblott-github changing this line to `(document.activeElement ? window).focus()` should do the right thing. It was a silly oversight, sorry.

We should make sure to guard against `document.activeElement` being `null`, since the [spec specifically says that it can happen](https://html.spec.whatwg.org/#focus-management-apis:document-8).